global/scripts/parallel.shlib

```diff
@@ -172,3 +185,73 @@ function par_addjob_stdin()
     par_jobinputarray+=("$jobinput")
 }
 
+#Add job to the current stage (FSLSUB, BUILTIN or NONE mode)
+#Inputs:
+#1. parallel mode
+#2. queue name for FSLSUB (local submit if empty)
+#3. command with arguments
+function par_add_job_to_stage()
+{
+    local PARALLEL_MODE=$1
+    shift
+
+    #FSLSUB queue name for FSLSUB mode, should be 'NONE' for other modes.
+    local fslsub_options fslsub_queue=$1
+    shift
+
+    local job submit_cmd cmd=("$@")
+
+    if [ "$PARALLEL_MODE" == "FSLSUB" ]; then #current job is submitted, and added to fsl_job_list
+        fslsub_options=""
+        submit_cmd=(fsl_sub)
+        if [[ -n "$fslsub_queue" ]]; then 
+            submit_cmd+=(-q "$fslsub_queue")
+        fi
+        if [[ -n "$par_fslsub_job_waitlist" ]]; then
+            submit_cmd+=(-j "$par_fslsub_job_waitlist")
+        fi
+        submit_cmd+=("${cmd[@]}")        
+        job=$("${submit_cmd[@]}")
+        par_fslsub_job_list+=("$job")
+    elif [ "$PARALLEL_MODE" == "BUILTIN" ]; then          #current job is added to the list of local parallel jobs        
+        par_addjob $cmd        
+    else #NONE                                          #current job is executed.
+        echo "launching, parallel mode=NONE, will wait to finish: ${cmd[*]}"
+        ${cmd} 1> ./"$par_global_scriptname.$$.$par_global_launch_count.o.log" 2> ./"$par_global_scriptname.$$.$par_global_launch_count.e.log"
+        par_global_launch_count=$((par_global_launch_count + 1))        
+    fi
```

Comment:
You could have used github to accept my suggestion, and then you wouldn't have missed these edits.
```suggestion
        par_addjob "${cmd[@]}"
    else #NONE                                          #current job is executed.
        echo "launching, parallel mode=NONE, will wait to finish: ${cmd[*]}"
        "${cmd[@]}" 1> ./"$par_global_scriptname.$$.$par_global_launch_count.o.log" 2> ./"$par_global_scriptname.$$.$par_global_launch_count.e.log"
        par_global_launch_count=$((par_global_launch_count + 1))
    fi
```

Reply:
Ok